Wordle.Api/Wordle.Api/Services/WordleService.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using System.Linq;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Services
+{
+  public class WordleService 
+  {
+    private readonly PlayerDbContext _context;
+
+    public WordleService(PlayerDbContext context)
+    {
+      _context = context;
+    }
+
+    public List<Player> GetTop10Scores()
+    {
+      return _context.Players.OrderByDescending(p => p.AverageAttempts).ThenBy(p => p.GameCount).Take(10).ToList();
+    }
+
+    public void AddOrUpdatePlayer(Player player)
```

Comment:
I like how you made this into one whole method instead of two separate. Do you think making them separate would make for easier testing and code structure?

Reply:
Possibly, though this implementation seems better to me since you don't have to make another method to check if the name is already in the database or not.